apps/keyboard/js/keyboard/layout_normalizer.js

```diff
@@ -34,6 +34,14 @@ LayoutKeyNormalizer.prototype._isSpecialKey = function(key) {
   return hasSpecialCode || key.keyCode <= 0;
 };
 
+LayoutKeyNormalizer.prototype._isButton = function(key) {
+  var SPECIAL_BUTTONS = [
+      KeyEvent.DOM_VK_RETURN
+  ];
+
+  return this._isSpecialKey(key) && SPECIAL_BUTTONS.indexOf(key.keyCode) !== -1;
```

Comment:
I assumed that only special keys should ever be considered as buttons (which is consistent with the past where `if(key.isSpecial} { ... } else { /* add role=key */ }`), and so the check there is meaningful to us.